language_bindings/rust/rust_build/src/lib.rs

```diff
@@ -0,0 +1,360 @@
+#![warn(missing_docs)]
+//#![warn(missing_doc_code_examples)]
+
+//!This crate is an example of calling Halide generated code from Rust.
+//!This crate also has an example of IIR-blur Halide app that get called from rust.
+//! IIR blur takes a image input, goes through the full Halide pipeline, and outputs a blurred image all using rust.
+//!
+
+use std::fs;
+use std::io::{ErrorKind, Result};
+use std::ops::Add;
+use std::path::PathBuf;
+use std::process::{Command, Output};
+///Required module for use in build.rs scripts
+///
+/// This Module will compile and bind a generator and create a Halide runtime.
+///
+//TODO update rust docs
+use std::{env, io};
+
+/// This builder must have a path to a complied Halide folder. It is used to build multiple generators quickly
+///
+///
+pub struct GenBuilder {
+    halide_path: PathBuf,
+    gen_path: PathBuf,
+
+    rs_output: PathBuf,
```

Comment:
The code below frequently calls `to_str().unwrap()` on these, and in some cases they are constructed directly from strings. If paths must be strings, it might be more robust and efficient to store a `String` from the start, avoiding all the conversions and possible panics.